### DISCUSSION TITLE
Shutdown button erases everything without warning

### DISCUSSION THREAD

**Discussion by kevadesu** [DustinBrett/daedalOS#279] [Feature Requests]
So basically I wanted to see what the power key did and it just cleared my session and had to upload things again. Is there any way to backup and restore things or atleast add a warning?

**Comment by DustinBrett** [comment]
Hey, thanks for the feedback. I will consider this for the future as a possible popup. Currently there is a tooltip that explains this will happen, but indeed it is a bit quick and can surprise people I suppose. As for saving the state, there is `/session.json` for some stuff but for files that have been written I currently have no way to backup this if it's reset.

![image](https://user-images.githubusercontent.com/16656318/223774101-bdad8eb6-14a6-443c-a246-54b22f671ad6.png)

**Comment by CorruptionVR** [comment]
Maybe add a popup/modal window saying that it will delete everything?

**Comment by kevadesu** [comment]
just asking, when is this gonna be fixed?

**Reply by DustinBrett** [reply]
I hadn't considered it something that needed fixing so it's been low on priority. But I will re-consider it if people are finding it annoying. I assumed people mostly don't click it and if they do then it just resets which I hadn't considered much of an issue.

**Comment by CorruptionVR** [comment]
hey, I had an idea. Maybe make the warning the "closing apps and shutting down" screen

**Comment by DustinBrett** [comment]
I will consider the Windows 10 shutdown dialog that you get if you try and Alt+F4 from the desktop.

![image](https://github.com/DustinBrett/daedalOS/assets/16656318/5d20717b-cd4a-4bcc-9baa-e716bfcc3c60)

**Reply by CorruptionVR** [reply]
I did not know that was a thing

**Comment by kevadesu** [comment]
That's epic, thanks for considering it
